cargo-workspaces/src/publish.rs

```diff
@@ -88,16 +88,21 @@ impl Publish {
 
             let name_ver = format!("{} v{}", name, version);
 
-            let mut index =
-                if let Some(publish) = pkg.publish.as_deref().and_then(|x| x.get(0)).as_deref() {
-                    let registry_url = cargo_config_get(
-                        &metadata.workspace_root,
-                        &format!("registries.{}.index", publish),
-                    )?;
-                    Index::from_url(&format!("registry+{}", registry_url))?
-                } else {
-                    Index::new_cargo_default()?
-                };
+            let mut index = if let Some(registry) = self.registry.as_deref() {
```

Comment:
You can do `self.registry.as_deref().or_else(...)` instead of creating two similar branches here.